server/websocket/src/server.rs

```diff
@@ -189,30 +116,38 @@ where
                             // Obtain the subgraph ID or name that we resolved the request to
                             let subgraph_id = subgraph_id.lock().unwrap().clone().unwrap();
 
+                            // Check if the subgraph is deployed
+                            match store2.is_deployed(&subgraph_id) {
+                                Err(_) |
+                                Ok(false) => {
+                                    error!(logger, "Failed to establish WS connection, subgraph not deployed";
+                                                "subgraph" => subgraph_id.to_string(),
+                                    );
+                                    return Ok(());
```

Comment:
Will returning here close the WS connection? I think that's desirable. And if we can it would be good to report a 'subgraph not found' type error back to client.

Reply:
Moved this up to reject the connection earlier with a 404.